aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/transfer/ProgressListener.kt

```diff
@@ -118,26 +134,35 @@ internal class DownloadProgressListener(
     private val transferStatusUpdater: TransferStatusUpdater
 ) : ProgressListener {
     private var bytesTransferredSoFar = 0L
+    private var progressUpdateSink = 0L
 
     init {
         bytesTransferredSoFar = transferRecord.bytesCurrent
     }
 
     override fun progressChanged(bytesTransferred: Long) {
         bytesTransferredSoFar += bytesTransferred
+        progressUpdateSink += bytesTransferred
         transferRecord.bytesCurrent = bytesTransferredSoFar
         if (transferRecord.bytesCurrent > transferRecord.bytesTotal) {
             transferRecord.bytesCurrent = transferRecord.bytesTotal
         }
-        updateProgress()
+        if (progressUpdateSink >=
+            min(transferRecord.bytesTotal / 100, transferRecord.bytesTotal - transferRecord.bytesCurrent)
+        ) {
+            updateProgress()
+            progressUpdateSink = 0L
+        }
     }
 
     private fun updateProgress() {
         transferStatusUpdater.updateProgress(
             transferRecord.id,
             transferRecord.bytesCurrent,
             transferRecord.bytesTotal,
-            true
+            true,
+            // writing progress to DB is not required for download operation, as current progress could be inferred from the file length
+            updateDB = false
```

Comment:
What about pausing and then resuming after a device restart? Is expectation that resume starts over?

Reply:
even on device restart file should not be deleted, incase if it is deleted then restart.

Reply:
I'm not referring to deletion. I'm more thinking of the scenario that download is 50% complete, device is restarted. Before this change, would the download resume at the midway point? With this change, if we don't keep track of download progress, it would start at the beginning.